src/lti_toolbox/launch_params.py

```diff
@@ -90,42 +100,63 @@
     "content_item_return_url",
     "data",
     "title",
-]
+}
 
-CONTENT_PARAMS_RESPONSE = [
+CONTENT_PARAMS_RESPONSE = {
     "content_items",
     "lti_errorlog",
     "lti_errormsg",
     "lti_log",
     "lti_msg",
-]
+}
 
-REGISTRATION_PARAMS = [
+REGISTRATION_PARAMS = {
     "reg_key",
     "reg_password",
     "tc_profile_url",
-]
+}
 
 LAUNCH_PARAMS = (
     CONTENT_PARAMS_REQUEST
-    + CONTENT_PARAMS_RESPONSE
-    + LAUNCH_PARAMS_CANVAS
-    + LAUNCH_PARAMS_LIS
-    + LAUNCH_PARAMS_OAUTH
-    + LAUNCH_PARAMS_RECOMMENDED
-    + LAUNCH_PARAMS_REQUIRED
-    + LAUNCH_PARAMS_RETURN_URL
-    + REGISTRATION_PARAMS
+    | CONTENT_PARAMS_RESPONSE
+    | LAUNCH_PARAMS_CANVAS
+    | LAUNCH_PARAMS_LIS
+    | LAUNCH_PARAMS_OAUTH
+    | LAUNCH_PARAMS_RECOMMENDED
+    | LAUNCH_PARAMS_REQUIRED
+    | LAUNCH_PARAMS_RETURN_URL
+    | REGISTRATION_PARAMS
 )
 
+SELECTION_PARAMS_REQUIRED = {
+    "lti_message_type",
+    "lti_version",
+    "accept_media_types",
+    "accept_presentation_document_targets",
+    "content_item_return_url",
+}
+SELECTION_PARAMS_SHOULD_NOT_BE_PASSED = {
+    "resource_link_id",
+    "resource_link_title",
+    "resource_link_description",
+    "launch_presentation_return_url",
+    "lis_result_sourcedid",
+}
+
+SELECTION_PARAMS = LAUNCH_PARAMS - SELECTION_PARAMS_SHOULD_NOT_BE_PASSED
+
 
-class LaunchParams(MutableMapping):
+class ParamsMixins(MutableMapping):
     """
-    Represents the params for a LTI launch request. Provides dict-like
-    behavior through the use of the MutableMapping ABC mixin.  Strictly
+    Represents the params for an LTI request. Provides dict-like
+    behavior through the use of the MutableMapping ABC mixin. Strictly
     enforces that params are valid LTI params.
     """
 
+    params_allowed: Set[str] = set()
+    params_required: Set[str] = set()
+    params_is_list: Set[str] = set()
+
     def __init__(self, *args, **kwargs):
```

Comment:
I was thinking of adding an extra check, on the LTI message type. Making sure LaunchParams has a `"basic-lti-launch-request"` value and SelectionParams a `"ContentItemSelectionRequest"` value.

Reply:
Yes!